MediaBrowser.Controller/MediaSegements/IMediaSegmentProvider.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using MediaBrowser.Model;
+using MediaBrowser.Model.MediaSegments;
+
+namespace MediaBrowser.Controller;
+
+/// <summary>
+/// Provides methods for Obtaining the Media Segments from an Item.
+/// </summary>
+public interface IMediaSegmentProvider
```

Comment:
our pattern is adding a `Supports(BaseItem item)` or a property to get a list of supported media types- maybe this should be added so we don't need to call every plugin for every item